File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
import java.util.*;

class Solution {
    public String solution(String s, String skip, int index) {
        Set<Character> skipSet = new HashSet<>();
        for (char c : skip.toCharArray()) {
            skipSet.add(c);
        }

        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            int count = 0;
            while (count < index) {
                c++;

                if (c > 'z') {
                    c = 'a';
                }

                if (skipSet.contains(c)) {
                    continue;
                }
                count++;
            }
            sb.append(c);
        }

        return sb.toString();
    }
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    arr = []

    for c in s:
        count = 0
        while count < index:
            c = chr((ord(c) - ord('a') + 1) % 26 + ord('a'))
            if c not in skip:
                count += 1

        arr.append(c)

    return ''.join(arr)